Clear-ComputeNode.ps1

```diff
@@ -0,0 +1,202 @@
+Param (
+    [Parameter(Mandatory = $false)] [String] $AdapterName = "Ethernet1",
+    [Parameter(Mandatory = $false)] [String] $ForwardingExtensionName = "vRouter forwarding extension",
+    [Parameter(Mandatory = $false)] [String] $VMSwitchName = "Layered Ethernet1",
+    [Parameter(Mandatory = $false)] [String] $ConfigAndLogDir = "C:\ProgramData\Contrail",
+    [Parameter(Mandatory = $false)] [String] $InstallationDir = "C:\Program Files\Juniper Networks"
+)
+
+function Invoke-ScriptBlockAndPrintErrors {
+    Param (
+        [Parameter(Mandatory=$true, Position=0)] [ScriptBlock] $ScriptBlock
+    )
+
+    try {
+        Invoke-Command $ScriptBlock
+    } catch {
+        $ErrorMessage += $_.Exception.Message
+        Write-Host "    This step failed with the following error message: $ErrorMessage"
+        Write-Host "    Trying to continue, but final result is unpredictable."
+    }
+}
+
+function Stop-ProcessIfExists {
+    Param ([Parameter(Mandatory = $true)] [String] $ProcessName)
+
+    $Proc = Get-Process $ProcessName -ErrorAction SilentlyContinue
+    if ($Proc) {
+        $Proc | Stop-Process -Force -PassThru | Wait-Process -ErrorAction Stop
+    }
+}
+
+function Remove-HNSNetworks {
+    Write-Host "Cleaning HNS state..."
+    Invoke-ScriptBlockAndPrintErrors {
+        Get-NetNat | Remove-NetNat -Confirm:$false
```

Comment:
Removing NetNat should be a separate function I think. It's only indirectly related to HNS network created by docker upon startup.